src/main/java/io/stargate/sgv2/jsonapi/service/cqldriver/executor/DefaultDriverExceptionHandler.java

```diff
@@ -1,36 +1,516 @@
+              m.put(
+                  "values",
+                  errFmtJoin(CqlPrintUtil.trimmedPositionalValues(statement), Object::toString));
+              m.put("blockForNodes", String.valueOf(exception.getBlockFor()));
+              m.put("receivedNodes", String.valueOf(exception.getReceived()));
+              m.put("dataPresent", String.valueOf(exception.wasDataPresent()));
+            }));
+  }
+
+  /**
+   * Same reasons as the {@link #handle(ReadFailureException)}, but with some different information
+   */
+  @Override
+  public RuntimeException handle(WriteFailureException exception) {
+    var uniqueReasons =
+        exception.getReasonMap().values().stream()
+            .distinct()
+            .map(RequestFailureReason::fromCode)
+            .map(Enum::name)
+            .toList();
+
+    return DatabaseException.Code.FAILED_WRITE_REQUEST.get(
+        errVars(
+            schemaObject,
+            exception,
+            m -> {
+              m.put("cql", CqlPrintUtil.trimmedCql(statement));
+              m.put(
+                  "values",
+                  errFmtJoin(CqlPrintUtil.trimmedPositionalValues(statement), Object::toString));
+              m.put("blockForNodes", String.valueOf(exception.getBlockFor()));
+              m.put("receivedNodes", String.valueOf(exception.getReceived()));
+              m.put("failedNodes", String.valueOf(exception.getNumFailures()));
+              m.put("writeType", String.valueOf(exception.getWriteType()));
+              m.put("failureReasons", errFmtJoin(uniqueReasons));
+            }));
+  }
+
+  @Override
+  public RuntimeException handle(WriteTimeoutException exception) {
+    return DatabaseException.Code.TIMEOUT_WRITING_DATA.get(
+        errVars(
+            schemaObject,
+            exception,
+            m -> {
+              m.put("cql", CqlPrintUtil.trimmedCql(statement));
+              m.put(
+                  "values",
+                  errFmtJoin(CqlPrintUtil.trimmedPositionalValues(statement), Object::toString));
+              m.put("blockForNodes", String.valueOf(exception.getBlockFor()));
+              m.put("receivedNodes", String.valueOf(exception.getReceived()));
+              m.put("writeType", String.valueOf(exception.getWriteType()));
+            }));
+  }
+
+  /**
+   * Factory function for creating a new instance of a DriverExceptionHandler.
+   *
+   * <p>Implementations of the interface should provide a factory function that can be called to
+   * create a new instance of the handler. This can just be the constructor for the default class.
+   * For example:
+   *
+   * <pre>
+   *   return new GenericOperation<>(attempts, pageBuilder, TableDriverExceptionHandler::new);
+   * </pre>
+   *
+   * See usage of the {@link
+   * io.stargate.sgv2.jsonapi.service.operation.tables.TableDriverExceptionHandler}
+   */
+  @FunctionalInterface
+  public interface Factory<T extends SchemaObject>
```

Comment:
Naming: this was confusing from the other end -- type defines not a factory, but something to call/apply to get/construct a factory. So maybe... `FactorySupplier`? (or `Provider`? not "FactoryFactory" tho :) ).

Not a big deal but just noticed as a minor readability concern.